### PR TITLE
add useful addresses body request param

### PR DESCRIPTION
**Pull request by juniormp**

### CODE REVIEW

File: src/routes/blockchains.js [hunk 1]
```diff
@@ -116,6 +116,8 @@ router.put(
             blockchain.transactionExplorerUrl,
           addressExplorerUrl:
             req.body.addressExplorerUrl ?? blockchain.addressExplorerUrl,
+          usefulAddresses:
```
[the remark this answers is not in the document]
Comment (juniormp) [reply src/routes/blockchains.js]: @tcoratger Makes sense.
I just pushed the new changes. 
Now we have 2 new routes to manage useful addresses 🚀

File: src/routes/blockchains.js [hunk 2]
```diff
@@ -154,4 +156,131 @@ router.delete(
   }
 );
 
+/* This is a post request to the blockchain route. It is using the createBlockchainValidator to
+validate the request body. It is also using the isRequired middleware to check if the user is logged
+in. If the user is not logged in, it will return a 401 error. If the user is logged in, it will
+check if the blockchain already exists. If it does not exist, it will create the blockchain. If it
+does exist, it will return a 404 error. */
+router.post('/', createBlockchainValidator, isRequired, async (req, res) => {
```
Comment (tcoratger) [inline src/routes/blockchains.js:164]: @juniormp What is the difference between this route and the first one of the  file? Why is it necessary to do a duplication here of the route to create a new blockchain?

File: src/routes/blockchains.js [hunk 3]
```diff
@@ -154,4 +156,131 @@ router.delete(
...
+    ) {
+      return res.status(400).send(validator);
+    }
+    const blockchain = await collection.findOne({
+      _id: new ObjectId(req.params.blockchainId),
+      usefulAddresses: {
+        $elemMatch: { contract: req.body.contract },
+      },
+    });
+    if (blockchain) {
```
Comment (tcoratger) [inline src/routes/blockchains.js:208]: @juniormp There is also the case where `req.params.blockchainId` is not found in the collection: in this case, the route must return a 404 error "Blockchain not found"

File: src/test/blockchains.test.js [hunk 4]
```diff
@@ -570,4 +574,105 @@ describe('Blockchains route', async function () {
...
   });
+  describe('POST useful address', () => {
+    describe('Core of the route', () => {
+      it('Should return 403 if no token is not provided', async function () {
+        const res = await chai
+          .request(app)
+          .post('/test/blockchains/useful-address/1234');
+        chai.expect(res).to.have.status(403);
+      });
+      it('Should create a new useful address with the proper fields', async function () {
```
Comment (tcoratger) [inline src/test/blockchains.test.js:585]: @juniormp can you test also the situation where `myContract1` already exists and so you are modifying the address with this call?

File: src/test/blockchains.test.js [hunk 5]
```diff
@@ -570,4 +574,105 @@ describe('Blockchains route', async function () {
...
+  });
+  describe('DELETE useful address', () => {
+    describe('Core of the route', () => {
+      it('Should return 403 if no token is not provided', async function () {
+        const res = await chai
+          .request(app)
+          .delete('/test/blockchains/useful-address/1234');
+        chai.expect(res).to.have.status(403);
+      });
+      it('Should return 404 if useful address not found', async function () {
```
Comment (tcoratger) [inline src/test/blockchains.test.js:634]: @juniormp Can you also test the situation where the blockchain itself is not found?

**Pull request merged**